### ISSUE TITLE
Phone verification error occurs

### ISSUE THREAD

**Issue by v4rchaem** [kenorb-contrib/tg#51]
phone number: +821012341234
*** 1638365853.511603 Notification API_64BIT_LOGIN_APP_OUTDATED_153: You are using an outdated app that is no longer supported. To access your messages, please update your app to the latest version.
*** 1638365853.512399 Incorrect phone number

**Comment by GeoIH** [comment]
Hey, v4rchaem, could you solve it?

**Comment by v4rchaem** [comment]
no, the problem continue

**Comment by GeoIH** [comment]
holy craps please @kenorb help us¡¡¡

**Comment by ulrickno94** [comment]
Damn, got the same error.

**Comment by andrew-phi** [comment]
Tg devs updated the protocol, even if you get tgcli running, it won't reveive any messages. This is a server-side issue and the client needs to be rewriteen.

**Comment by burkass** [comment]
@andrew-phi yes correct, currently my bot using cli not working. But no issue when sending message. And it looks like login with cli will not working.

**Comment by sijumanuel** [comment]
same issue

**Comment by juliobm** [comment]
same here

**Comment by abitmore** [comment]
FWIW error messages when running `telegram-cli -vvvv`:
```
phone number: +xxxxxx
 *** 1638631722.367240 Sent query #7037869656658957312 of size 204 to DC 2
 *** 1638631722.461673 update 0x382dd3e4 (check=0)
 *** 1638631722.461915 Notification API_64BIT_LOGIN_APP_OUTDATED_222: You are using an outdated app that is no longer supported. To access your messages, please update your app to the latest version.
 *** 1638631722.462198 error for query 'send code' #7037869656658957312: #400 UPDATE_APP_TO_LOGIN
 *** 1638631722.462386 Incorrect phone number
```

**Comment by uriid1** [comment]
+1 
same issue

**Comment by abitmore** [comment]
According to the doc https://core.telegram.org/method/auth.sendCode, our code for login (`auth.sendCode#768d5f4d`) is on layer (version) 23.
```
auth.sentCode#efed51d9 phone_registered:Bool phone_code_hash:string send_call_timeout:int is_password:Bool = auth.SentCode;
auth.sentAppCode#e325edcf phone_registered:Bool phone_code_hash:string send_call_timeout:int is_password:Bool = auth.SentCode;
---functions---
auth.sendCode#768d5f4d phone_number:string sms_type:int api_id:int api_hash:string lang_code:string = auth.SentCode;
```

 The latest layer (version) is 133 and it requires 64-bit IDs for User/Chat

```
auth.sentCode#5e002502 flags:# type:auth.SentCodeType phone_code_hash:string next_type:flags.1?auth.CodeType timeout:flags.2?int = auth.SentCode;
---functions---
auth.sendCode#a677244f phone_number:string api_id:int api_hash:string settings:CodeSettings = auth.SentCode;
```

An upgrade to the `tgl` lib is required.

By the way, due to some issues I forgot, the default `tgl` lib in this repository (https://github.com/kenorb-contrib/tgl) stopped working for me, so I switched to https://github.com/majn/tgl, and now it no longer works either.

**Comment by denimjeans** [comment]
Same issue here, I belive we are now forced to use the TDlib. 
https://core.telegram.org/tdlib
Hope someone will implement this soon.

**Comment by staoran** [comment]
Currently we can use Telethon, it can achieve the same effect.

**Comment by burkass** [comment]
@staoran can it store number and Sent that number with stored username?

**Comment by staoran** [comment]
@burkass Yes, it is as convenient as CLI. For more details, please see its documentation: https://docs.telethon.dev/en/latest/index.html

**Comment by ViktorKravtsov** [comment]
@staoran is there compliled exe-file for using on windows? Or instruction how to do this?

**Comment by staoran** [comment]
@ViktorKravtsov It is full platform, it is a python package, you can do anything with python.

**Comment by burkass** [comment]
Issue occur again this time after put the code and same error came. Already use -q

![image](https://user-images.githubusercontent.com/5495419/152670099-4fb90680-b014-4c64-989a-529120b57335.png)

**Comment by zqhong** [comment]
@staoran 

```
telethon.errors.rpcerrorlist.ChatWriteForbiddenError: You can't write in this chat (caused by SendMessageRequest)
```

When I send a message to a regular user using telethon, I get an error like the one above.

**Comment by staoran** [comment]
Sorry, I haven't encountered this problem, you can go to the following
places to take a look:
https://stackoverflow.com/questions/65588256/how-to-send-message-to-my-channel-using-telethon
https://www.google.com/search?q=You+can%27t+write+in+this+chat+(caused+by+SendMessageRequest)&oq=You+can%27t+write+in+this+chat+(caused+by+SendMessageRequest)&aqs=edge..69i57&sourceid=chrome&ie=UTF-8

zqhong ***@***.***> 于2022年4月17日周日 15:56写道：

**Comment by zqhong** [comment]
@staoran 

Thank you, it's a problem with my code. The following code is similar to your code and works fine. Where the app_id and api_hash are applied at [https://my.telegram.org/auth](https://my.telegram.org/auth) this website.

```python
from telethon.sync import TelegramClient

if __name__ == '__main__':
    api_id = 12345
    api_hash = 'xxx'
    session_name = "session_name"

    client = TelegramClient(session_name, api_id, api_hash)
    client.start()
    client.send_message("@bot_name", '/sign')

```
